registrator.go

```diff
@@ -172,6 +173,10 @@ func main() {
 			go b.Add(msg.ID)
 		case "die":
 			go b.RemoveOnExit(msg.ID)
+		case "kill":
+			if *deregisterOnStop {
+				go b.RemoveOnExit(msg.ID)
+			}
 		}
```

Comment:
For the above suggestion, note that we can only determine the signal in Docker API version 1.22, which corresponds to Docker version 1.10, so this might not be a viable solution if we need to support previous Docker versions.